### PR TITLE
chore: remove require usage for aws-sdk imports to use bundled dep

### PR DESCRIPTION
**Pull request by canavandl**
# Description
The Node18 Lambda runtime doesn't provide the v2 aws-sdk, which is used by the modules' service libraries. In order to unblock the Node16 (upcoming EOL) -> Node18 migration, this change causes the service code to use the bundled aws-sdk library instead of the runtime-provided one.

node18 migration: See: https://github.com/aws/aws-connected-device-framework/issues/96
v2 aws-sdk is also upcoming EOL and the service code needs to be migrated to v3. There's an issue to track here: https://github.com/aws/aws-connected-device-framework/issues/135

## Type of change

- [ ] *Bug fix (non-breaking change which fixes an issue)*
- [ ] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [X] *Refactor* (existing code being refactored)
- [ ] *This change includes a documentation update*

# Submission Checklist

- [X] CI dry-run passing
- [X] Integration tests passing locally
- [ ] Change logs generated

### CODE REVIEW

File: source/packages/services/assetlibrary/src/config/.env.defaults [hunk 1]
```diff
@@ -23,7 +23,7 @@ CACHE_TYPES_TTL=30
 
 SUPPORTED_API_VERSIONS=application/vnd.aws-cdf-v1.0+json,application/vnd.aws-cdf-v2.0+json
 
-AUTHORIZATION_ENABLED=true
```
Comment (canavandl) [inline source/packages/services/assetlibrary/src/config/.env.defaults:26]: unrelated change to temporarily resolve https://github.com/aws/aws-connected-device-framework/issues/131

**Review by anish-kunduru — APPROVED** [review]
Approved, as per the intended description.

However, I think we also need to update the relevant `package.json` files to add the SDK as a dependency. We will also need to modify the [deploy.json](https://github.com/aws/aws-connected-device-framework/blob/main/source/common/config/rush/deploy.json#L145) to ensure it isn't excluded from the lambda bundle, as that is no longer provided in the v18 environment.

Finally, we might wish to bump up the `MemorySize` on the lambdas to mitigate the impact this will have on cold starts.

**Comment by canavandl** [comment]
>I think we also need to update the relevant package.json files to add the SDK as a dependency
For the service libraries, there's no required change as the aws-sdk package is bundled into the output zipfile via esbuild. For the client libraries, I just made aws-sdk a normal dependency instead of a dev dep, so it'll be installed however all of the other dependencies are packages (hopefully also via esbuild).
>We will also need to modify the [deploy.json](https://github.com/aws/aws-connected-device-framework/blob/main/source/common/config/rush/deploy.json#L145)
I just removed the `dependenciesToExclude` section, since the service lambda code is esbuild-ed into a single file now instead of us uploading a node_modules directory
>Finally, we might wish to bump up the MemorySize on the lambdas to mitigate the impact this will have on cold starts.
AFAIK this change shouldn't affect cold starts as the necessary aws-sdk code is now bundled into the lambda code

File: source/common/config/rush/deploy.json [hunk 2]
```diff
@@ -140,109 +140,5 @@
...
-      "projectName": "@awssolutions/cdf-lambda-invoke",
-      "dependenciesToExclude": ["aws-sdk"]
-    },
-    {
-      "projectName": "@awssolutions/cdf-logger",
-      "dependenciesToExclude": ["aws-sdk"]
-    },
-    {
-      "projectName": "@awssolutions/cdf-assetlibrary",
-      "dependenciesToExclude": ["aws-sdk"]
```
Comment (canavandl) [inline source/common/config/rush/deploy.json:185]: for all of the service packages, this was previously a no-op as the `aws-sdk` code was bundled into the lambda zip via esbuild.

**Review by anish-kunduru — APPROVED** [review]

**Review by BenjiTheC — APPROVED** [review]
